From a2ee204fc2b77c1edff47f9bce5c2d9baa8f46b1 Mon Sep 17 00:00:00 2001
From: mytja <mitja@severkar.eu>
Date: Sat, 17 May 2025 15:48:46 +0200
Subject: [PATCH 16/16] feat(fw/b): set rounded corners

I honestly have no idea why people want to set rounded corners, but
here you go, I guess.
---
 core/java/android/view/RoundedCorners.java | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/core/java/android/view/RoundedCorners.java b/core/java/android/view/RoundedCorners.java
index 1f841bf53364..c5a5fe1849a4 100644
--- a/core/java/android/view/RoundedCorners.java
+++ b/core/java/android/view/RoundedCorners.java
@@ -31,6 +31,7 @@ import android.content.res.TypedArray;
 import android.graphics.Rect;
 import android.os.Parcel;
 import android.os.Parcelable;
+import android.os.SystemProperties;
 import android.util.DisplayUtils;
 import android.util.Pair;
 import android.view.RoundedCorner.Position;
@@ -197,6 +198,10 @@ public class RoundedCorners implements Parcelable {
         } else {
             radius = res.getDimensionPixelSize(R.dimen.rounded_corner_radius);
         }
+        int syspropRadius = SystemProperties.getInt("persist.sys.phh.rounded_corners_padding", -1);
+        if (syspropRadius >= 0) {
+            radius = syspropRadius;
+        }
         array.recycle();
         // For devices with round displays (e.g. watches) that don't otherwise provide the rounded
         // corner radius via resource overlays, we can infer the corner radius directly from the
@@ -225,6 +230,10 @@ public class RoundedCorners implements Parcelable {
         } else {
             radius = res.getDimensionPixelSize(R.dimen.rounded_corner_radius_top);
         }
+        int syspropRadius = SystemProperties.getInt("persist.sys.phh.rounded_corners_padding", -1);
+        if (syspropRadius >= 0) {
+            radius = syspropRadius;
+        }
         array.recycle();
         return radius;
     }
@@ -248,6 +257,10 @@ public class RoundedCorners implements Parcelable {
         } else {
             radius = res.getDimensionPixelSize(R.dimen.rounded_corner_radius_bottom);
         }
+        int syspropRadius = SystemProperties.getInt("persist.sys.phh.rounded_corners_padding", -1);
+        if (syspropRadius >= 0) {
+            radius = syspropRadius;
+        }
         array.recycle();
         return radius;
     }
-- 
2.34.1

